utils/swift_build_support/swift_build_support/clang.py

```diff
@@ -0,0 +1,90 @@
+# swift_build_support/clang.py - Detect host machine's Clang -*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+#
+# Find the path to a Clang executable on the host machine that is most
+# suitable for building Swift.
+#
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import
+
+import collections
+import platform
+import subprocess
+
+from . import xcrun
+from .which import which
+
+
+# A named tuple consisting of two paths:
+# 1. 'cc' is the path to a program used for compiling C.
+# 2. 'cxx' is the path to a program used for compiling C++.
+CompilerExecutable = collections.namedtuple('CompilerExecutable', 'cc cxx')
+
+
+def _freebsd_release_date():
+    """
+    Return the release date for FreeBSD operating system on this host.
+    If the release date cannot be ascertained, return None.
+    """
+    try:
+        # For details on `sysctl`, see:
+        # http://www.freebsd.org/cgi/man.cgi?sysctl(8)
+        return int(subprocess.check_output(
+            ['sysctl', '-n', 'kern.osreldate']).rstrip())
+    except subprocess.CalledProcessError:
+        return None
+
+
+def _first_clang(suffixes):
+    """
+    Return a CompilerExecutable with the first available versions of clang
+    and clang++, searching in the order of the given suffixes.
+
+    If no Clang executables are found, return None.
+    """
+    for suffix in suffixes:
+        cc_path = which('clang{}'.format(suffix))
+        cxx_path = which('clang{}'.format(suffix))
```

Comment:
This should say `clang++` -- I'm getting strange linker errors because CMake is trying to link C++ programs with the C compiler.